Claims manager

Ant-Nazif opened the pull request:
--------------------------------------------------------------------
 * Created claims manager layout
 * Created move to category layout
 * Memoized checkIfClaimSelected
 * Changed updateCategoryForClaim to take a list of claims
 * Made small adjustments to the seed attestations
 * Display the name of the attested claim insted of attestationId
 * Added functionality to move claim/claims into different categories
--------------------------------------------------------------------

Ant-Nazif committed:
CHECKPOINT / Claims manager layout

--------------------------------------------------------------------
 MAIN
 * Created claims manager layout
 * Created move to category layout
--------------------------------------------------------------------

kkuna committed:
COMPLETED / Show/hide claims

--------------------------------------------------------------------
 MAIN
 * Added functionality to show/hide claims on claim manager
 * Added functionality to show/hide hidden claims on claim category details
--------------------------------------------------------------------

Ant-Nazif committed:
CHECKPOINT/Claim manager UI fixes

Ant-Nazif committed:
CHECKPOINT/Removed unused code

kkuna committed:
COMPLETED / Move claims into different categories

--------------------------------------------------------------------
 MAIN
 * Added functionality to move claim/claims into different categories
--------------------------------------------------------------------
 OTHER
 * Changed claimCategories and claims seed data structure
--------------------------------------------------------------------

Ant-Nazif committed:
CHECKPOINT/Claim manager UI fixes

Ant-Nazif committed:
CHECKPOINT/Switched show/hide icon

Ant-Nazif committed:
CHECKPOINT/UI fixes

kkuna committed:
COMPLETED / Claim manager code cleanup

--------------------------------------------------------------------
 MAIN
 * Memoized checkIfClaimSelected
 * Changed updateCategoryForClaim to take a list of claims
--------------------------------------------------------------------

kkuna committed:
CHECKPOINT / Attestations renaming

--------------------------------------------------------------------
 MAIN
 * Made small adjustments to the seed attestations
 * Display the name of the attested claim insted of attestationId
--------------------------------------------------------------------

Ant-Nazif committed:
CHECKPOINT/Fix search categories by claims

Ant-Nazif committed:
CHECKPOINT/Fix search personal info

hanzah commented on src/actions/actionCreators.js:
```diff
@@ -76,6 +76,12 @@ import {
+  UPDATE_SELECTED_CLAIMS,
+  CLEAR_SELECTED_CLAIMS,
+  UPDATE_CATEGORY_FOR_CLAIMS,
+  MOVE_CLAIMS_TO_CATEGORY,
```
Why do we keep all action creators in one file. We can split this file up based on domain, for example claims, categories, identity etc.
You can even maintain the import path just by creating an actionCreators directory and moving the exports into the index.js of it.

hanzah commented on src/actions/actionCreators.js:
```diff
@@ -447,3 +458,27 @@ export const setNewCategory = (category) => ({
+  payload: { claim, value },
+});
+
+export const updateSelectedClaims = (claim) => ({
```
If you name it in plural, you would expect that the action creator takes an array of claims as a parameter and not one claim.

hanzah commented on src/actions/actionCreators.js:
```diff
@@ -447,3 +458,27 @@ export const setNewCategory = (category) => ({
+});
+
+export const clearSelectedClaims = () => ({
+  type: CLEAR_SELECTED_CLAIMS,
```
Maintain the footprint/signature of an action. Even if the payload is just an empty object, it should be there.

hanzah commented on src/actions/actionCreators.js:
```diff
@@ -447,3 +458,27 @@ export const setNewCategory = (category) => ({
+  payload: { claims, categoryId },
+});
+
+export const moveClaimsToCategory = (targetCategory) => ({
```
What claims are you moving? If it's something from the state, make is visible in the name of the action creator.

hanzah commented on src/containers/Identity/PersonalInfo/ClaimCategoryDetails/View.js:
```diff
@@ -10,16 +10,23 @@ const ClaimCategoryDetails = (props) => {
+      <CheckBox
+        checked={showHiddenClaims}
+        title="Show hidden claims"
+        onPress={() => setShowHiddenClaims(!showHiddenClaims)}
```
If you are only using this for this kind of use case. You can name the action toggleHiddenClaims and do all the logic in the saga.

hanzah commented on src/containers/Identity/PersonalInfo/ClaimManager/MoveIntoCategory/styles.js:
```diff
@@ -0,0 +1,13 @@
+    padding:16,
+  },
+  text:{
+    color:'#b5b5b5',
```
Think about extracting the colors in a top theme styles file, so we can adjust it in one place. Of course if this is a color that is dependent of the theme.

hanzah commented on src/containers/Identity/PersonalInfo/ClaimManager/View.js:
```diff
@@ -0,0 +1,89 @@
+
+  const checkIfClaimIsSelected = useCallback((claim) => selectedClaims.includes(claim), [selectedClaims]);
+
+  const getCategotyName = (name) => {
```
This is an util function, move it outside of the component.

Ant-Nazif replied:
done

hanzah commented on src/containers/Identity/PersonalInfo/ClaimManager/View.js:
```diff
@@ -0,0 +1,89 @@
+                checked={checkIfClaimIsSelected(claims.get(claim))}
+                onPress={() => selectClaim(claims.get(claim, IMap()))}
+                title={claims.getIn([claim, 'name'])}
+                containerStyle={{ backgroundColor:'white' }}
```
Move into styles

Ant-Nazif replied:
done

hanzah commented on src/containers/Identity/PersonalInfo/ClaimManager/View.js:
```diff
@@ -0,0 +1,89 @@
+              </TouchableOpacity>
+              <TouchableOpacity onPress={() => toggleClaimVisibility(claims.get(claim))}>
+                {claims.getIn([claim, 'hidden']) ? <Icon name={Platform.OS === 'ios' ? 'ios-eye-off' : 'md-eye-off'} size={24} />
+                  : <Icon name={Platform.OS === 'ios' ? 'ios-eye' : 'md-eye'} size={24} />
```
These can be separate components, so you can reuse them whenever you need them in the app.
Size can be the prop.

<Icon name={Platform.OS === 'ios' ? 'ios-eye-off' : 'md-eye-off'} size={24} />
<Icon name={Platform.OS === 'ios' ? 'ios-eye' : 'md-eye'} size={24} />

Ant-Nazif replied:
done

hanzah commented on src/containers/Identity/PersonalInfo/ClaimManager/styles.js:
```diff
@@ -0,0 +1,66 @@
+  },
+  button: {
+    backgroundColor:
+            '#b5b5b5',
```
Same thing and code styling. The linter should catch this.

hanzah commented on src/containers/Identity/PersonalInfo/View.js:
```diff
@@ -72,17 +72,30 @@ const PersonalInfo = (props) => {
+
+      if (itemDataName.includes(search)) {
+        return itemDataName.includes(search);
+      }
```
if (itemDataName.includes(search)) {
        return true;
      }

Ant-Nazif replied:
done

hanzah commented on src/containers/Identity/PersonalInfo/View.js:
```diff
@@ -72,17 +72,30 @@ const PersonalInfo = (props) => {
 
   const updateSearch = (searchValue) => {
     const newData = claimCategories.filter((claimCategory) => {
```
Please add a comment what this function is doing and how. Think about moving it into a util function just pass into it whatever it needs.

Ant-Nazif replied:
done

hanzah commented on src/sagas/identity.js:
```diff
@@ -170,14 +185,27 @@ function * handleAddNewCategory(action) {
   yield call(updateClaimCategoryStorage);
 }
 
+function * handleMoveClaims(action) {
```
Start thinking about how the API will be integrated an error handling, so your code can anticipate it easily.

hanzah commented on src/selectors/identity.js:
```diff
@@ -21,28 +21,44 @@ export const selectClaimCategoriesObj = (state) => state.identity.getIn(['person
 );
 
-const selectClaims = (state) => state.identity.getIn(['personalInformation', 'claims', 'byId'], IMap());
+export const selectClaimsObj = (state) => state.identity.getIn(['personalInformation', 'claims'], IMap());
```
Think about a more descriptive name, maybe something like selectClaimsReducerState or something down that line.
Obj is just to ambiguous.

hanzah commented on src/selectors/identity.js:
```diff
@@ -21,28 +21,44 @@ export const selectClaimCategoriesObj = (state) => state.identity.getIn(['person
+export const selectClaimsToDisplay = createSelector(
+  [selectClaimsByCategoryId, selectShowHiddenClaims],
+  (selectedClaims, showHiddenClaims) => {
+    if (!showHiddenClaims) {
```
Why the reverse, you are just adding an additional operation
if(showHiddenClaims) {
  return selectedClaims
}
....

hanzah commented on src/selectors/identity.js:
```diff
@@ -21,28 +21,44 @@ export const selectClaimCategoriesObj = (state) => state.identity.getIn(['person
+  [selectClaimsByCategoryId, selectShowHiddenClaims],
+  (selectedClaims, showHiddenClaims) => {
+    if (!showHiddenClaims) {
+      return selectedClaims.filter((claim) => claim.get('hidden', '') === false);
```
selectedClaims.filter((claim) => !claim.get('hidden'));

hanzah commented on src/utils/asyncStore/identityStorage.js:
```diff
@@ -89,30 +89,39 @@ export const updateClaimCategories = async (updatedCategories) => {
 };
 
-export const getIdentities = async () => {
+export const storeSeedClaims = async (identityId) => {
```
Please add a comment what this function is doing and all the cases it is handling.

hanzah commented on src/utils/asyncStore/identityStorage.js:
```diff
@@ -143,8 +152,9 @@ export const updateIdentities = async (updatedIdentities) => {
 export const updateClaims = async (updatedClaims) => {
```
You should not be doing these calculations in the storage repository. Same as storeSeedClaims.
The business logic needs to go into the sagas. 
These repository functions should be simple stupid, just retreiving and storing what you pass into them.

Ant-Nazif committed:
CHECKPOINT / PR fixes regarding comments

kkuna committed:
UPDATE / Claims manager

--------------------------------------------------------------------
 MAIN
 * Moved action identity app action creators to their own files
 * Generate unique ids for seed attestations
 * Generate unique ids for seed claims
 * Persist pinned attestation in async storage
 * Moved business logic from async storage setters/getters to sagas
 * Changed the toggle functions for showing hidden claims, claim categories etc.
--------------------------------------------------------------------
